api/src/test/java/org/pmiops/workbench/api/ProfileControllerTest.java

```diff
@@ -769,6 +774,39 @@ public void testUpdateNihToken_serverError() {
     profileController.updateNihToken(new NihToken().jwt("test"));
   }
 
+  @Test
+  public void testSyncTraining() throws Exception {
+    List<BadgeDetails> badgeDetail = new ArrayList<>();
+    Timestamp time = new Timestamp(12543);
+    BadgeDetails badge = new BadgeDetails();
+    badge.setName("All of us badge");
+    badge.setDateexpire("12543");
+    badgeDetail.add(badge);
+    when(complianceTrainingService.getMoodleId(PRIMARY_EMAIL)).thenReturn(12);
+    when(complianceTrainingService.getUserBadge(12)).thenReturn(badgeDetail);
+
+    createUser();
+
+    profileController.syncTrainingStatus();
+    verify(complianceTrainingService).getMoodleId(PRIMARY_EMAIL);
+    assertThat(userDao.findUserByEmail(PRIMARY_EMAIL).getTrainingExpirationTime()).isEqualTo(time);
+    assertThat(userDao.findUserByEmail(PRIMARY_EMAIL).getTrainingExpirationTime()).isNotNull();
+
+  }
+
+  @Test
+  public void testSynTrainingWithNoBadge() throws Exception {
+    List<BadgeDetails> badgeDetail = new ArrayList<>();
+
+    when(complianceTrainingService.getMoodleId(PRIMARY_EMAIL)).thenReturn(12);
+    when(complianceTrainingService.getUserBadge(12)).thenReturn(badgeDetail);
+
+    createUser();
+
+    profileController.syncTrainingStatus();
+    assertThat(userDao.findUserByEmail(PRIMARY_EMAIL).getTermsOfServiceCompletionTime()).isNull();
+    assertThat(userDao.findUserByEmail(PRIMARY_EMAIL).getTrainingExpirationTime()).isNull();
+  }
```

Comment:
It would be useful to have a test case for the scenario where the Moodle API returns a NOT_FOUND response for getMoodleId, since we expect that to happen fairly often (e.g. when an AoU researcher creates their account and lands on the /access page).